39.md

```diff
@@ -13,26 +13,34 @@ Nostr protocol users may have other online identities such as usernames, profile
 ## `i` tag on a metadata event
 
 A new optional `i` tag is introduced for `kind 0` metadata event contents in addition to name, about, picture fields as included in [NIP-01](https://github.com/nostr-protocol/nips/blob/master/01.md):
-```json
+
+```jsonc
 {
+  "kind": 0,
   "tags": [
     ["i", "github:semisol", "9721ce4ee4fceb91c9711ca2a6c9a5ab"],
     ["i", "twitter:semisol_public", "1619358434134196225"],
-    ["i", "mastodon:bitcoinhackers.org/@semisol", "109775066355589974"]
-    ["i", "telegram:1087295469", "nostrdirectory/770"]
+    ["i", "mastodon:bitcoinhackers.org/@semisol", "109775066355589974"],
+    ["i", "telegram:1087295469", "nostrdirectory/770"],
+    ["i", "openpgp4fpr:1A04E0F1A78D982BD8885B7EB325A9C5F70849D0", "LS0tLS1CRUdJTiBQR1AgTUVTU0FHRS0tLS0tCgpvd0did012TXdDVzJXWFhsMGU4Y25oY1lUeTlJWWtpVGw0bHpxbFFvemt6UHk4eExWeWpKeUN4V3lFMHRMazVNClQxWHdWRWpPejB2TExNb0ZDaWVXUUxnbFJmazVRRzZxUWtGUlpsbGlTYXBDZG1xbFFscCtFVmpNTDcrNHBFaWgKb0RRcEp6TVpMSkVIWkJ1V3A1a1VsS1lWbHlhbkZwbVdKVnFrV3hZWUZHV1pwdVJsbEtXbFpwaWxXSlFicEp1bApKbFltcGhhWXBtUVVGSnNWRmFjWG14aW5wR2RZY25XVXNqQ0ljVEhJaWlteVNMRTgrTGk4ZDRiMmpZN29PcGdQCldKbEF6bWZnNGhTQWlTU0pNVExjL2E3YmRtZUx3S0dMZWs2NkRTcDYvNmRmMzl6Nk5pN1E1dmcxLytCU0NRY1oKUm9ZOUwvN2NhWW05WnRZN2FkNm1GMy95ak83OVZqbWhZZmF4bm1mNlFaYldDM0lNQUE9PQo9ZXkyagotLS0tLUVORCBQR1AgTUVTU0FHRS0tLS0tCg", "LS0tLS1CRUdJTiBQR1AgUFVCTElDIEtFWSBCTE9DSy0tLS0tCgptRE1FWmdIZE94WUpLd1lCQkFIYVJ3OEJBUWRBNnBaZ2RzbUR2aVJGNDVDYys5TUJiK3VkUURZZ2ZHRjZrQkJVCk1CeUd4S2EwRjJaeVlXNTZZWEFnUEY5QVpuSmhibnBoY0M1amIyMCtpSk1FRXhZS0FEc1dJUVFhQk9EeHA0MlkKSzlpSVczNnpKYW5GOXdoSjBBVUNaZ0hkT3dJYkF3VUxDUWdIQWdJaUFnWVZDZ2tJQ3dJRUZnSURBUUllQndJWApnQUFLQ1JDekphbkY5d2hKMEw5R0FRQ3VFUFF0aW9EdGlKaWNVSGw4S0RqU0dmRU1laEFWdjZsTGJuZDBTRjN6Ci9BRCtQZWJCcFk3YkwvakY4QnViaU8rVm9nbUhsa2NGZ3h5V2ZEMDYwMDAwb0FvPQo9OWRBUAotLS0tLUVORCBQR1AgUFVCTElDIEtFWSBCTE9DSy0tLS0tCg"]
   ],
-  ...
+  // ...
 }
 ```
 
-An `i` tag will have two parameters, which are defined as the following:
-1. `platform:identity`: This is the platform name (for example `github`) and the identity on that platform (for example `semisol`) joined together with `:`.
-2. `proof`: String or object that points to the proof of owning this identity.
+An `i` tag will have two parameters and an optional third parameter, which are defined as the following:
+1. `type:identity`: This is the platform name (for example `github`) or the linked cryptographic identity type (for example `openpgp4fpr`), and the identity on that platform (for example `semisol`) or public key fingerprint (for example `1A04E0F1A78D982BD8885B7EB325A9C5F70849D0`) joined together with `:`.
+2. `proof`: String that points to the proof of owning this identity or is the proof itself (crytographic signature).
+3. The full public key in the case of a cryptographic identity.
```

Comment:
OpenPGP keys can be very large, depending on the number of subkeys, annotations, signatures, etc – mine is 38kB and I have seen much larger.
Rather than embedding the whole key, it would make more sense to provide a URL where it can be downloaded.
Either way, I think it should be optional, and if absent clients can fetch it from keys.openpgp.org or another default keyserver of their choice.

Reply:
I think at such a size, and for the sake of simplicity, the key should be elsewhere. Maybe an optional bloomserver hash is fine, but raw key may break clients if the key is so large.

Is there a bloomserver uri scheme?

Reply:
What is a bloomserver?

Reply:
@caesar My God, PGP is painful. Is there any way to compress and throw away any subkeys and annotations and just leave the meat of it?  I want to avoid having to ping keyservers as much as possible, they are a security hole.

Reply:
Hmm, I'm not sure. I don't really see a problem with just providing an external URI for the key though, all the existing proofs are provided via external links anyway. (As for keyservers, I agree, but it's part of the PGP system and not really our problem to solve in Nostr in my opinion.)

Reply:
Okay. For people who think their key is too heavy (we could suggest a threshold, TBD) we should recommend adding multiple keyservers to mitigate both DNS and keyserver operator rugging risks

Reply:
+1 on the threshold. Good compromise.

Reply:
@caesar It was late, but I meant blossom server: https://github.com/hzrd149/blossom

It gives some DNS issue resistance, but I guess so does listing keyservers. I just like the idea of referencing and downloading files by their hash.

Maybe it is a bad idea, the wiki suggests that key servers are built more for purpose and have evolved over time: https://en.m.wikipedia.org/wiki/Key_server_(cryptographic)
(See problems with keyservers section)

Reply:
I will add the ability to specify URLs as a fallback when public key material is too heavy. Blossom is great but we should not be prescriptive: if people share a Blossom-like URL then it can be assumed as such.

Reply:
`gpg --edit-key` shows different options to reduce size like `clean`, `minimize`, `delphoto` etc